### PR TITLE
Rating: phase 0.5. `TestingType`, `best_execution` in `lnk_contest_project`

### PR DESCRIPTION
**Pull request by petertrr**
* Change `ExecutionType` to `TestingType`, pass `testingType` from the frontend
* Add columns for best execution and best score in `lnk_contest_project`

What's left for phase 1:
* Move `score` into `Execution`
* Add creation of `LnkProjectExecution`s

First part of #1115

### CODE REVIEW

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/RunExecutionController.kt [hunk 1]
```diff
@@ -63,11 +64,13 @@ class RunExecutionController(
     /**
      * @param request incoming request from frontend
      * @param authentication
+     * @param testingType type for this execution
      * @return response with ID of created [Execution]
      */
     @PostMapping("/trigger")
     fun trigger(
         @RequestBody request: RunExecutionRequest,
+        @RequestParam testingType: TestingType,
```
Comment (nulls) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/RunExecutionController.kt:73]: @akuleshov7 , should we allow `/re-trigger` for contests ?

File: save-orchestrator/src/test/kotlin/com/saveourtool/save/orchestrator/controller/agents/AgentsControllerTest.kt [hunk 2]
```diff
@@ -75,7 +75,7 @@ class AgentsControllerTest {
     fun `should build image, query backend and start containers`() {
         val project = Project.stub(null)
         val execution = Execution.stub(project).apply {
-            type = ExecutionType.STANDARD
+            type = TestingType.PRIVATE_TESTS
```
Comment (nulls) [inline save-orchestrator/src/test/kotlin/com/saveourtool/save/orchestrator/controller/agents/AgentsControllerTest.kt:78]: just a note: `ExecutionType.GIT` is `TestingType.PRIVATE_TESTS` and `ExecutionType.STANDARD` is `TestingType.PUBLIC_TESTS`, but in tests and stub you changed it in another way

**Review by nulls — APPROVED** [review]

File: save-orchestrator/src/test/kotlin/com/saveourtool/save/orchestrator/controller/agents/AgentsControllerTest.kt (hunk 2 above)
Comment (petertrr) [reply save-orchestrator/src/test/kotlin/com/saveourtool/save/orchestrator/controller/agents/AgentsControllerTest.kt:78]: Thanks, must've confused them in some places. Will double check

File: save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/Execution.kt [hunk 3]
```diff
@@ -156,7 +156,7 @@ class Execution(
             status = ExecutionStatus.RUNNING,
             testSuiteIds = null,
             batchSize = 20,
-            type = ExecutionType.GIT,
+            type = TestingType.PUBLIC_TESTS,
```
Comment (petertrr) [inline save-cloud-common/src/jvmMain/kotlin/com/saveourtool/save/entities/Execution.kt:159]: For some reason we had different types in stubs for `Execution` and `ExectuionDto`. Will set both to `PUBLIC`

**Pull request merged**